Classes/ContentRepository/Service/NodeService.php

```diff
@@ -86,36 +92,43 @@ public function getNodeFromContextPath($contextPath, Site $site = null, Domain $
         $nodePath = $nodePathAndContext['nodePath'];
         $workspaceName = $nodePathAndContext['workspaceName'];
         $dimensions = $nodePathAndContext['dimensions'];
+        $siteNodeName = $site ? $site->getNodeName() : explode('/', $nodePath)[2];
```

Comment:
this `explode('/'` will be impossible to upmerge to neos 9 ... but we wont need it there i guess as `currentSite` will not be needed so fine with me ;)